Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -39,34 +47,69 @@ internal static class ProfileManager
-                
-                string arguments = System.Environment.ExpandEnvironmentVariables(string.Format("start {0} /attach:{1} /loadAgent:4EA90761-2248-496C-B854-3C0399A591A4;DiagnosticsHub.CpuAgent.dll  /scratchLocation:%LOCAL_EXPANDED%\\Temp", profilingSessionId, processId));
 
-                var profileProcessResponse = await ExecuteProfilingCommandAsync(arguments, tracer);
+                if (iisProfiling)
+                {
+                    var profileResponseStatus = await StartIISSessionAsync(processId, profilingSessionId, tracer);
 
-                if (profileProcessResponse.StatusCode != HttpStatusCode.OK)
+                    if (profileResponseStatus.StatusCode != HttpStatusCode.OK)
+                    {
+                        return profileResponseStatus;
+                    }
+                }
+                else
                 {
-                    return profileProcessResponse;
+                    string arguments = System.Environment.ExpandEnvironmentVariables(string.Format("start {0} /attach:{1} /loadAgent:{2};DiagnosticsHub.CpuAgent.dll  /scratchLocation:%LOCAL_EXPANDED%\\Temp", profilingSessionId, processId, DiagnosticsHubAgentGuid));
+
+                    var profileProcessResponse = await ExecuteProfilingCommandAsync(arguments, tracer);
+
+                    if (profileProcessResponse.StatusCode != HttpStatusCode.OK)
+                    {
+                        return profileProcessResponse;
+                    }
                 }
 
                 // This may fail if we got 2 requests at the same time to start a profiling session
                 // in that case, only 1 will be added and the other one will be stopped.
-                if (!_profilingList.TryAdd(processId, new ProfileInfo(profilingSessionId)))
+                if (!_profilingList.TryAdd(processId, new ProfileInfo(profilingSessionId, iisProfiling)))
                 {
                     tracer.TraceWarning("A profiling session was already running for process {0}, stopping profiling session {1}", processId, profilingSessionId);
-                    await StopProfileInternalAsync(processId, profilingSessionId, true, tracer);
+                    await StopProfileInternalAsync(processId, profilingSessionId, true, tracer, iisProfiling);
                     return new ProfileResultInfo(HttpStatusCode.OK, string.Empty);
                 }
 
                 tracer.Step("started session id: {0} for pid: {1}", profilingSessionId, processId);
 
-                EnsureIdleTimer();
+                EnsureIdleTimer(iisProfiling);
 
                 return new ProfileResultInfo(HttpStatusCode.OK, string.Empty);
             }
         }
 
-        internal static async Task<ProfileResultInfo> StopProfileAsync(int processId, ITracer tracer = null)
+        private static async Task<ProfileResultInfo> StartIISSessionAsync(int processId, int profilingSessionId, ITracer tracer = null)
```

Comment:
How about some comments explaining what's going on here?
e.g. Start a session with the "custom etw provider agent", add in the "detailed tracing agent", turn on the IISWWWServer provider.